EventFilter/L1TRawToDigi/src/implementations_stage2/EtSumPacker.cc

```diff
@@ -36,30 +36,42 @@ namespace stage2 {
       edm::Handle<EtSumBxCollection> etSums;
       event.getByToken(static_cast<const CommonTokens*>(toks)->getEtSumToken(), etSums);
 
-      uint32_t et_word = 0;
-      uint32_t ht_word = 0;
-      uint32_t met_word = 0;
-      uint32_t mht_word = 0;
-
+      std::vector<uint32_t> load;
+      int nBx = 0;
       for (int i = etSums->getFirstBX(); i <= etSums->getLastBX(); ++i) {
+
+	 uint32_t et_word = 0;
+	 uint32_t ht_word = 0;
+	 uint32_t met_word = 0;
+	 uint32_t mht_word = 0;
+
          for (auto j = etSums->begin(i); j != etSums->end(i); ++j) {
 	   uint32_t word = std::min(j->hwPt(), 0xFFF);
 	   if ((j->getType()==l1t::EtSum::kMissingEt) || (j->getType()==l1t::EtSum::kMissingHt))
 	     word = word | ((j->hwPhi() & 0xFF) << 12);
 	   
-	   if (j->getType()==l1t::EtSum::kTotalEt)   et_word = word;
-	   if (j->getType()==l1t::EtSum::kTotalHt)   ht_word = word;
-	   if (j->getType()==l1t::EtSum::kMissingEt) met_word = word;
-	   if (j->getType()==l1t::EtSum::kMissingHt) mht_word = word;
+	   if (j->getType()==l1t::EtSum::kTotalEt)       et_word  |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFP0)   et_word  |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kTotalHt)       ht_word  |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFM0)   ht_word  |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kMissingEt)     met_word |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFP1)   met_word |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kMissingHt)     mht_word |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFM1)   mht_word |= (word << 28);
```

Comment:
I agree these need to be changed.  Perhaps @jimbrooke or @bundocka want to
do this?  I originally made these bug fixes so that I could test mutliboard
uGT readout, but this is more their territory.  If they want me to do it, I
can; but I would feel more comfortable it they put in the changes.

  -Brian

On Thu, Jun 30, 2016 at 8:33 AM, Andrea Bocci <notifications@github.com>
wrote:

> In EventFilter/L1TRawToDigi/src/implementations_stage2/EtSumPacker.cc
> <https://github.com/cms-l1t-offline/cmssw/pull/333#discussion_r69123226>:
>
> >           for (auto j = etSums->begin(i); j != etSums->end(i); ++j) {
> >  	   uint32_t word = std::min(j->hwPt(), 0xFFF);
> >  	   if ((j->getType()==l1t::EtSum::kMissingEt) || (j->getType()==l1t::EtSum::kMissingHt))
> >  	     word = word | ((j->hwPhi() & 0xFF) << 12);
> >  	
> > -	   if (j->getType()==l1t::EtSum::kTotalEt)   et_word = word;
> > -	   if (j->getType()==l1t::EtSum::kTotalHt)   ht_word = word;
> > -	   if (j->getType()==l1t::EtSum::kMissingEt) met_word = word;
> > -	   if (j->getType()==l1t::EtSum::kMissingHt) mht_word = word;
> > +	   if (j->getType()==l1t::EtSum::kTotalEt)       et_word  |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFP0)   et_word  |= (word << 28);
> > +	   if (j->getType()==l1t::EtSum::kTotalHt)       ht_word  |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFM0)   ht_word  |= (word << 28);
> > +	   if (j->getType()==l1t::EtSum::kMissingEt)     met_word |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFP1)   met_word |= (word << 28);
> > +	   if (j->getType()==l1t::EtSum::kMissingHt)     mht_word |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFM1)   mht_word |= (word << 28);
>
> I would suggest changing the enum to a enum class, and the if into a
> switch, which will help the compiler generate warnings if one introduces
> new elements into the enum but fails to handle them.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-l1t-offline/cmssw/pull/333/files/9d205b3c737d05eaebd4b301c6fe7750570b2117#r69123226>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AElaBv9issVRE23kdJzLPmPO1EHfk9b4ks5qQ7d9gaJpZM4IlnuI>
> .
>



-- 
----------------------------------------
B. Winer